backend/api/views/quiz.py

```diff
@@ -1,80 +1,77 @@
 
-    def get(self, request: Request) -> JsonResponse:
-        sort_order = get_sort_order_from_request(request, self.__quizzes)
-        return JsonResponse(data=self.__quizzes.find(sort=sort_order))
+class QuizList(QuizApiView):
+    """LIST all quizzes and CREATE new quiz."""
 
-    def post(self, request: Request) -> JsonResponse:
+    def get(self, request: Request):
+        sort_order = get_sort_order(request.query_params, self._quizzes)
+        return JsonResponse(data=self._quizzes.find(sort=sort_order))
+
+    def post(self, request: Request):
         quiz_data = request.data
-        quiz_data["id"] = get_new_id(self.__quizzes)
+        quiz_data["id"] = get_new_id()
 
         try:
-            self.__validator.validate(quiz_data)
+            self._validator.validate(quiz_data)
         except InvalidData as error:
             return JsonResponse(
                 data={"message": f"Invalid JSON: {str(error)}"},
                 status=status.HTTP_400_BAD_REQUEST,
             )
         else:
-            insertion_result = self.__quizzes.insert_one(quiz_data)
-            inserted_quiz = self.__quizzes.find_one(
+            insertion_result = self._quizzes.insert_one(quiz_data)
+            inserted_quiz = self._quizzes.find_one(
                 {"_id": insertion_result.inserted_id}
             )
             return JsonResponse(data=inserted_quiz, status=status.HTTP_201_CREATED)
 
 
-class QuizDetails(APIView):
-    __quizzes = client.collection("quizzes")
-    __validator = SchemaValidator("quizzes")
+class QuizDetail(QuizApiView):
+    """GET, PATCH and DELETE a single quiz."""
 
-    def get(self, request: Request, pk: str) -> JsonResponse:
-        if request.method == "GET":
-            quiz = self.__quizzes.find_one({"id": pk})
+    def get(self, request: Request, pk: str):
+        quiz = self._quizzes.find_one({"id": pk})
 
-            if quiz is None:
-                raise Http404("There is no quiz matching your query.")
-            else:
-                return JsonResponse(data=quiz)
+        if quiz is None:
+            raise Http404("There is no quiz matching your query.")
+        else:
+            return JsonResponse(data=quiz)
 
     def patch(self, request: Request, pk: str) -> JsonResponse:
-        if request.method == "PATCH":
-            quiz_data = request.data
+        quiz_data = request.data
 
-            try:
-                self.__validator.validate(quiz_data)
-            except InvalidData as error:
-                return JsonResponse(
-                    data={"message": f"Invalid JSON: {str(error)}"},
-                    status=status.HTTP_400_BAD_REQUEST,
-                )
-            else:
-                updated_quiz = self.__quizzes.find_one_and_update(
-                    {"id": pk},
-                    {"$set": quiz_data},
-                    # If the document is not already in collection, insert it
-                    upsert=True,
```

Comment:
As I said in your previous PR, `PATCH` should only modify an instance. If you had `PUT`, then `upsert=True` would've made sense

Reply:
Yes you're right 😉 I didn't pay enough attention when modifying.